boards/arm/cxd56xx/spresense/src/cxd56_ioctl.c

```diff
@@ -91,6 +92,22 @@ int board_ioctl(unsigned int cmd, uintptr_t arg)
         }
         break;
 #endif
+
+#ifdef CONFIG_CXD56_SDIO
+      /* CMD:           BOARDIOC_SDCARD_SETNOTIFYCB
+       * DESCRIPTION:   Set a callback function pointer to SDCard driver
+       *                to notify when card status is changed.
+       * ARG:           Callback function.
+       * CONFIGURATION: CONFIG_BOARDCTL & CONFIG_CXD56_SDIO
+       * DEPENDENCIES:  Board logic must provide board_app_initialization
+       */
+
+      case BOARDIOC_SDCARD_SETNOTIFYCB:
+        {
+          ret = board_sdcard_set_state_cb(arg);
```

Comment:
I think passing function pointers from app to kernel is not the best thing. Why such functionality is needed?

Reply:
Same question, could we use automount instead?

Reply:
Please consider enabling `FS_AUTOMOUNTER_DRIVER` and using a `ioctl(FIOC_NOTIFY)` instead of direct callback

Reply:
I think that such improvements should be done in another PR.